sktime/classification/_base.py

```diff
@@ -0,0 +1,516 @@
+                            in datatypes._check.py
+    """
+
+    _tags = {
+        "X_inner_mtype": "numpy3D",  # which type do _fit/_predict accept, usually
+        # this is either "numpy3D" or "nested_univ" (nested pd.DataFrame). Other
+        # types are allowable, see datatypes/panel/_registry.py for options.
+        "capability:multivariate": False,
+        "capability:unequal_length": False,
+        "capability:missing_values": False,
+        "capability:train_estimate": False,
+        "capability:contractable": False,
+        "capability:early_prediction": False,
+        "capability:multithreading": False,
+    }
+
+    def __init__(self):
+        self.classes_ = []
+        self.n_classes_ = 0
+        self.fit_time_ = 0
+        self._class_dictionary = {}
+        self._threads_to_use = 1
+        self._series_length = 0
+        self._X_metadata = {}
+        super(BaseClassifier, self).__init__()
+
+    def fit(self, X, y) -> BaseEstimator:
+        """Fit time series classifier to training data.
+
+        Parameters
+        ----------
+        X : np.ndarray (2d or 3d array of shape (n_instances, series_length) or shape
+            (n_instances, n_dimensions, series_length)) or pd.DataFrame (where each
+            column is a dimension, each cell is a pd.Series (any number of dimensions,
+            equal or unequal length series))
+            or of any other supported Panel data structures, for list of
+            supported, see MTYPE_REGISTER_PANEL in datatypes/panel/_registry.py
+        y : 1D np.array, of shape [n_instances] - class labels for fitting
+            indices correspond to instance indices in X
+
+        Returns
+        -------
+        self :  Fitted estimator.
+
+        Notes
+        -----
+        Converts X to type self.get_tag("X_inner_mtype"), creates a fitted model by
+        calling _fit that updates attributes ending in "_" and sets is_fitted to True.
+        """
+        start = int(round(time.time() * 1000))
+        # convenience conversions to allow user flexibility:
+        # if X is 2D array, convert to 3D, if y is Series, convert to 1D-numpy
+        X, y = _internal_convert(X, y)
+        #  Get data characteristics
+        self._X_metadata = _check_classifier_input(X, y)
+        missing = self._X_metadata["has_nans"]
+        multivariate = not self._X_metadata["is_univariate"]
+        unequal = not self._X_metadata["is_equal_length"]
+        # Check this classifier can handle these three characteristics
+        self._check_capabilities(missing, multivariate, unequal)
+        # Convert data as dictated by the classifier tag self.get_tag("X_inner_mtype")
+        X = self._convert_X(X)
+        # Store series length for checking in predict
+        if self.get_tag("X_inner_mtype") == "numpy3D":
+            self._series_length = self._find_series_length(X)
+        else:
+            self._series_length = 0
```

Comment:
but the series could be of equal length even if it is not of `numpy3D` mtype?

Reply:
if it is an important property for `predict` series and `fit` seris to have the same length, should we make this a metadata field that is returned by the checks?

Reply:
> Ok, I think there are two serious blockers:
> 
> * the duplication `base`/`_base` which is caused by MrSEQL. Let's park this PR until we have removed MrSEQL and that is no longer necessary. Let's not have hacks around key modules, especially not ones that duplicate files! I'd be really really hesitant moving that state into a release!

ok, thats fair. Pehaps I should just deprecate MrSEQL? 

> * we should discuss use of `abc` at one of the developer meetings. I feel the design point around optional abstract methods unresolved, e.g., `predict_proba`. If we park until the C code is deprecated, we should have ample time.

I dont agree that this should block the PR. I am not changing existing code structure with this PR, predict_proba is the same is were before. For sure consider other uses/designs for ABC, but please dont shoehorn it onto this PR. ABC is already used in benchmarking, clustering and distances.